src/function-builder-detectors/zisi.js

```diff
@@ -0,0 +1,67 @@
+const path = require('path')
+const process = require('process')
+
+const { zipFunction, zipFunctions } = require('@netlify/zip-it-and-ship-it')
+const makeDir = require('make-dir')
+
+const { getPathInProject } = require('../lib/settings')
+const { NETLIFYDEVERR } = require('../utils/logo')
+
+const bundleFunctions = ({ config, sourceDirectory, targetDirectory, updatedPath }) => {
+  // If `updatedPath` is truthy, it means we're running the build command due
+  // to an update to a file. If that's the case, we run `zipFunction` to bundle
+  // that specific function only.
+  if (updatedPath) {
+    return zipFunction(updatedPath, targetDirectory, {
+      archiveFormat: 'none',
+      config,
+    })
+  }
+
+  return zipFunctions(sourceDirectory, targetDirectory, {
+    archiveFormat: 'none',
+    config,
+  })
+}
+
+// The function configuration keys returned by @netlify/config are not an exact
+// match to the properties that @netlify/zip-it-and-ship-it expects. We do that
+// translation here.
+const normalizeFunctionsConfig = (functionsConfig = {}) =>
+  Object.entries(functionsConfig).reduce(
+    (result, [pattern, config]) => ({
+      ...result,
+      [pattern]: {
+        externalNodeModules: config.external_node_modules,
+        ignoredNodeModules: config.ignored_node_modules,
+        nodeBundler: config.node_bundler === 'esbuild' ? 'esbuild_zisi' : config.node_bundler,
+      },
+    }),
+    {},
+  )
+
+const getTargetDirectory = async ({ log }) => {
+  const targetDirectory = path.resolve(getPathInProject(['functions-serve']))
+
+  try {
+    await makeDir(targetDirectory)
+  } catch (error) {
+    log(`${NETLIFYDEVERR} Could not create directory: ${targetDirectory}`)
+
+    process.exit(1)
```

Comment:
Not sure whether to `exit` or let the error bubble up?

Reply:
To exit with an error we use `oclif` `error` utility, see https://github.com/netlify/cli/blob/f881b668de15bf5a9ec6988e5ce543caa7e22df8/src/commands/functions/create.js#L214

Regardless, I think this can be a warning as `netlify dev` will be partially useful?

Reply:
We won't be able to serve the functions if we can't write to the directory, though.

Reply:
We currently warn on function builder errors, so I think we can remove the `try/catch` here and let the CLI handle it:
https://github.com/netlify/cli/blob/f5b26d2295f46975194fc4c984e7695727b02029/src/utils/serve-functions.js#L409

Reply:
Changed in https://github.com/netlify/cli/pull/2094/commits/78f4ec0c541d8d21d7c3f2d42596c08aff87ef95 to use the `errorExit` property already being passed to the functions server. What do you think?

Reply:
Oh, I think we commented at the same time! 😅 

Just checking that you've seen https://github.com/netlify/cli/commit/78f4ec0c541d8d21d7c3f2d42596c08aff87ef95.